test/e2e/upgrade_test.go

```diff
@@ -206,3 +168,31 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
 
 	return err
 }
+
+// deployWindowsWorkloadAndTester tests if the Windows Webserver deployment is available.
+// This is achieved by creating a Job object that continuously curls the webserver every 5 seconds.
+// returns a tearDown func that must be executed to cleanup resources
+func (tc *testContext) deployWindowsWorkloadAndTester() (func(), error) {
+	// create a Windows Webserver deployment
+	deployment, err := tc.deployWindowsWebServer("win-webserver", nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating Windows Webserver deployment for upgrade test")
+	}
+	// create a clusterIP service which can be used to reach the Windows webserver
+	intermediarySVC, err := tc.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating service for deployment %s", deployment.Name)
```

Comment:
By return `nil` here you will end up not deleting the deployment and leak resources. How are you going to handle that?
Yes, it will get cleaned up on cluster deletion but given we are attempting a clean up, we should do that consistently. If you are not doing so for any reason, then it needs to be called here in a comment.

Reply:
Good catch, we should do that consistently and cleanup the already created objects. I missed that edge case, will adjust.